src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1138,6 +1138,36 @@ private AuthenticationResult acquireTokenLocalCall(final CallbackHandler callbac
     private boolean promptUser(PromptBehavior prompt) {
         return prompt == PromptBehavior.Always || prompt == PromptBehavior.REFRESH_SESSION;
     }
+    
+    
+    //checkRedictUri() checks the redirectUri before sending the request
+    //if the redirectUri from the client does not match the valid redirectUri
+    // the client app would not jump to the login page
+    //redirectUri format %PREFIX://%PACKAGE_NAME/%SIGNATURE   
+    private boolean checkRedictUri(CallbackHandler callbackHandle, final AuthenticationRequest request) throws UnsupportedEncodingException{
+    	
+    	String errMsg = new String();    	
+    	String inputUri = request.getRedirectUri();
+    	PackageHelper helper = new PackageHelper(mContext);
+    	
+    	if(inputUri == null){
+    		errMsg = "the redirectUri is null";    		
+    	}else if(!inputUri.split("://")[0].equals(AuthenticationConstants.Broker.REDIRECT_PREFIX)){
+    		errMsg = "the redirectUri is not starting with the valid redict prefix " + AuthenticationConstants.Broker.REDIRECT_PREFIX;    		
+    	}else if((inputUri.split("://").length > 1 && !inputUri.split("://")[1].split("/")[0].equals(URLEncoder.encode(mContext.getPackageName(), AuthenticationConstants.ENCODING_UTF8))) || inputUri.split("://").length == 1 ){
```

Comment:
i guess we can simply check if the given redirect uri contains the expected package name and signature. Can you also log out the expected package name and signature when it's wrong.